ipatests/test_integration/test_sudo.py

```diff
@@ -24,22 +24,43 @@
     clear_sssd_cache, get_host_ip_with_hostmask, modify_sssd_conf)
 
 
-class TestSudo(IntegrationTest):
-    """
-    Test Sudo
-    http://www.freeipa.org/page/V4/Sudo_Integration#Test_Plan
-    """
+class TestSudoBase(IntegrationTest):
     num_clients = 1
     topology = 'line'
 
     @classmethod
     def install(cls, mh):
-        super(TestSudo, cls).install(mh)
+        super().install(mh)
 
         cls.client = cls.clients[0]
         cls.clientname = cls.client.run_command(
             ['hostname', '-s']).stdout_text.strip()
 
+    def list_sudo_commands(self, user, raiseonerr=False, verbose=False,
+                           password=None):
+        clear_sssd_cache(self.client)
+        sudo_args = '-ll' if verbose else '-l'
+        stdin_text = None
+        if password is None:
+            sudo_args += ' -n'
+        else:
+            sudo_args += ' -S'
+            stdin_text = password + '\n'
+        return self.client.run_command(
+            ['su', '-', user, '-c', 'sudo %s' % sudo_args],
```

Comment:
You are still construction a string here. Did you try to pass in a list?

I'm picky here because this kind of code is a common security problem. It's a potential shell injection vector, something like a SQL injection but way worse. This code is not yet vulnerable, because you control all input variables. However the next person may change it inadvertently.

```
args = ['su', '-', user, '-c', 'sudo']
if verbose:
    args.append('-ll')
else:
    args.append('-l')
if password is not None:
    args.append('-S')
    stdin_text = password + '\n'
else:
    args.append('-n')
    stdin_text = None
...
```

Reply:
@tiran 
from `man su`:
`
 -c, --command=command
              Pass command to the shell with the -c option.
`
"command" must be a single parameter, it does not accept additional arguments.
"-ll", "-S", "-n" are arguments for "sudo" bu in your example strings  will be consumed but "su"